Plugins/Public/alley/PlayerRestrictions.cpp

```diff
@@ -1199,8 +1200,61 @@ void __stdcall ReqAddItem(unsigned int iArchID, char const *Hardpoint, int count
 		PrintUserCmdText(iClientID, L"Triggered on add equip");
 	}
 	*/
+
+	returncode = DEFAULT_RETURNCODE;
+	if (reverseTrade[iClientID] == true)
+	{
+		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+	}
+}
+
+void __stdcall GFGoodBuy(struct SGFGoodBuyInfo const &gbi, unsigned int client)
+{
+	returncode = DEFAULT_RETURNCODE;
+	if (false == SCI::CanBuyItem(gbi.iGoodID, client)) {
+		const GoodInfo* gi = GoodList::find_by_id(gbi.iGoodID);
+		switch (gi->iType) {
+			case GOODINFO_TYPE_COMMODITY: {
+				PrintUserCmdText(client, L"ERR Your ship can't load this cargo");
+				break;
+			}
+			case GOODINFO_TYPE_EQUIPMENT: {
+				PrintUserCmdText(client, L"ERR Your ship can't use this equipment");
```

Comment:
should we say 'use' or something more like 'store'? since this is more about purchasing rather than mounting

Reply:
You can't buy equipment you don't have a free slot to immediately mount it into, so I find this one fitting.